### PR TITLE
Merge throughput/zn_sub_thr_stream into z_sub_thr

### PR DESCRIPTION
**Pull request by william-wyliang**
This PR is to merge zn_sub_thr_stream of throughput into z_sub_thr.
A new option --no-callback is added to retain the behavior of zn_sub_thr_stream.

- [x]  verified with the following commands for the original z_sub_thr
    `$ ./target/debug/z_sub_thr -s my-test -m peer -l tcp/localhost:7447 -p 8 -n now`
    `$ ./target/debug/z_put_thr -m peer -l tcp/localhost:7447 -p 8`
- [x]  verified with the following commands for the merged functions of zn_sub_thr
    `$ ./target/debug/z_sub_thr -s my-test -m peer -l tcp/localhost:7447 -p 8 -n now --no-callback`
    `$ ./target/debug/z_sub_thr -s my-test -m peer -l tcp/localhost:7447 -p 8 -n now --declare_expr --no-callback`
    `$ ./target/debug/z_put_thr -m peer -l tcp/localhost:7447 -p 8`

### CODE REVIEW

**Review by eddie9712 — APPROVED** [review]
LGTM

File: throughput/src/bin/z_sub_thr.rs [hunk 1]
```diff
@@ -52,6 +52,10 @@ struct Opt {
     /// declare a numerical Id for the subscribed key expression
     #[clap(long = "declare_expr")]
```
Comment (YuanYuYuan) [inline throughput/src/bin/z_sub_thr.rs]: Maybe we can directly follow the convention of command-line arguments.

By changing the writing to below,

```rust    
    #[clap(long)]
    use_expr: bool,

    /// do not use callback for subscriber
    #[clap(long)]
    no_callback: bool,
```
we can call the program like `z_sub_thr --no-callback --declare-expr`.

Comment (william-wyliang) [reply throughput/src/bin/z_sub_thr.rs]: Agree. Will update.

Comment (YuanYuYuan) [reply throughput/src/bin/z_sub_thr.rs]: @william-wyliang  I have a small suggestion. Although you can specify the explicit name of a long argument, the default behavior is to change `no_callback` in Rust to `--no-callback` while using command-line arguments. So I think we can omit the term `#[clap(long = "...")]`. A simple `#[clap(long)]` is enough. Plus I think the word `use-expr` is better than `declare-expr` while we specify command-line arguments.

BTW, I have approved this PR. Feel free to merge it once you finished this modification. 😃

**Review by YuanYuYuan — APPROVED** [review]

File: throughput/src/bin/z_sub_thr.rs (hunk 1 above)
Comment (william-wyliang) [reply throughput/src/bin/z_sub_thr.rs]: Another PR has been submitted for this.

**Pull request merged**